editor/src/components/editor/component-button.tsx

```diff
@@ -0,0 +1,216 @@
+
+  const popupEnabled = selectedViews.length > 0
+
+  const [isOpen, setIsOpen] = React.useState(false)
+  const toggleOpen = React.useCallback(() => {
+    setIsOpen(!isOpen)
+  }, [isOpen])
+
+  const closeAndEatEvent = React.useCallback(
+    (e: MouseEvent) => {
+      setIsOpen(false)
+      e.stopPropagation()
+      e.preventDefault()
+    },
+    [setIsOpen],
+  )
+
+  enum editContext {
+    ELEMENT = 0,
+    COMPONENT_INSTANCE,
+    COMPONENT,
+    COMPONENT_CHILD,
+  }
+
+  const [currentEditContext, setCurrentEditContext] = React.useState(editContext.COMPONENT)
+
+  // TODO replace this with a function that enters focus mode
+  const TODOREPLACEMEcycleEditContext = () => {
+    const nextEditContext = (currentEditContext + 1) % 4
+    setCurrentEditContext(nextEditContext)
+  }
+
+  const getEditContextStyle = (): React.CSSProperties => {
+    if (currentEditContext === editContext.COMPONENT) {
+      return {
+        color: UtopiaTheme.color.component.value,
+        backgroundColor: UtopiaTheme.color.component.shade(10).value,
+        stroke: UtopiaTheme.color.component.fileNameFragment,
+      }
+    } else if (currentEditContext === editContext.COMPONENT_CHILD) {
+      return {
+        color: UtopiaTheme.color.componentChild.value,
+        backgroundColor: UtopiaTheme.color.componentChild.shade(10).value,
+        stroke: UtopiaTheme.color.componentChild.fileNameFragment,
+      }
+    } else if (currentEditContext === editContext.COMPONENT_INSTANCE) {
+      return {
+        background: UtopiaTheme.color.secondaryBackground.value,
+        color: UtopiaTheme.color.neutralForeground.value,
+        stroke: 'black',
+      }
+    } else {
+      return {
+        background: UtopiaTheme.color.secondaryBackground.value,
+        color: UtopiaTheme.color.neutralForeground.value,
+        stroke: 'black',
+        opacity: 0.5,
+        pointerEvents: 'none',
+      }
+    }
+  }
+
+  return (
+    <div
+      role='compositeButton'
+      style={{
+        position: 'relative',
+        display: 'flex',
+        alignItems: 'stretch',
+        height: UtopiaTheme.layout.inputHeight.default,
+        flexBasis: 130,
+      }}
+    >
+      <FlexRow
+        role='button'
+        // TODO replace me with the right function call from above
+        onClick={TODOREPLACEMEcycleEditContext}
+        css={{
+          flexGrow: 1,
+          flexShrink: 1,
+          overflow: 'hidden',
+          borderTopLeftRadius: UtopiaTheme.inputBorderRadius,
+          borderBottomLeftRadius: UtopiaTheme.inputBorderRadius,
+          gap: 8,
+          paddingLeft: 4,
+          cursor: 'pointer',
+          transition: 'background-color .1s ease-in-out',
+          ...getEditContextStyle(),
+          '&:hover': {
+            filter: 'brightness(1.02  )',
+          },
+          '&:active': {
+            filter: 'brightness(1.03)',
+          },
+        }}
+      >
+        {/* TODO replace me with the real icon */}
```

Comment:
is this TODO for the next PR ?

Reply:
Yes. We basically need the info from the focus mode work, so we can show the correct icon, label, colour, and status based on the select element(s).